gap.go

```diff
@@ -376,7 +376,10 @@ func (buf *rawAdvertisementPayload) reset() {
 // before the call) from the advertisement options. It returns true if it fits,
 // false otherwise.
 func (buf *rawAdvertisementPayload) addFromOptions(options AdvertisementOptions) (ok bool) {
-	buf.addFlags(0x06)
+	// do not add flags for non-connectable advertisements
+	if options.AdvertisementType != AdvertisingTypeNonConnInd {
```

Comment:
Do you want `AdvertisingTypeScanInd` here too? I.e.  `NONCONNECTABLE_SCANNABLE_UNDIRECTED`

Reply:
I did not make that assumption, but that might be true?

Probably best to handle that with another PR that makes it consistent for all platforms.